File: webflux-dialogue/src/main/java/com/study/webflux/rag/application/monitoring/port/ConversationMetricsPort.java
package com.study.webflux.rag.application.monitoring.port;

public interface ConversationMetricsPort {
	void recordConversationIncrement();

	void recordConversationReset();

	void recordQueryLength(int length);

	void recordResponseLength(int length);

	void recordConversationCount(long count);
}

File: backend/monitoring/src/main/kotlin/com/miyou/app/monitoring/port/ConversationMetricsPort.kt
package com.miyou.app.monitoring.port

interface ConversationMetricsPort {
    fun recordConversationIncrement()

    fun recordConversationReset()

    fun recordQueryLength(length: Int)

    fun recordResponseLength(length: Int)

    fun recordConversationCount(count: Long)

    fun recordFormatViolation()
}
